andreeapetcu/assignment3/Dictionary.java

```diff
@@ -0,0 +1,49 @@
+import java.util.List;
+
+/**
+ * The dictionary is implemented as a trie
+ * O(n) complexity for checking the existence of a word/ inserting a new word,
+ * where n is the length of the word
+ */
+
+public class Dictionary {
+
```

Comment:
I'm not sure adding a Dictionary class wrapping the Trie is really necessary, but you've done it cleanly with good javadoc so I can't criticise.

If you were to say "this is one possible implementation" then I might have expected an interface Dictionary, and this to be called TrieDictionary. But now I am really nitpicking. Don't worry too much, this is good.